engine/src/main/java/org/terasology/rendering/dag/nodes/ShadowMapNode.java

```diff
@@ -81,6 +80,8 @@ public void initialise() {
 
         addDesiredStateChange(new BindFBO(SCENE_SHADOW_MAP));
         addDesiredStateChange(new SetViewportSizeOf(SCENE_SHADOW_MAP));
+        addDesiredStateChange(new Clear(true, true, false));
```

Comment:
I don't think clearing operations are state changes. It's more of a degenerate type of rendering: you fill the buffer(s) with one color rather than a pretty picture. So, I'd leave clearing statements within the Node.process() methods and eventually verify and if they are actually useful. I feel that some are unnecessary rather than redundant.